src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Runtime/CSharpForwardCancellationTokenToAsyncMethods.Fixer.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.NetCore.Analyzers.Runtime;
+
+namespace Microsoft.NetCore.CSharp.Analyzers.Runtime
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp)]
+    public sealed class CSharpForwardCancellationTokenToAsyncMethodsFixer : ForwardCancellationTokenToAsyncMethodsFixer
+    {
+        private static bool IsCancellationTokenParameter(ParameterSyntax parameter) =>
+            parameter.Type is IdentifierNameSyntax typeIdentifier && typeIdentifier.Identifier.ValueText.Equals(CancellationTokenName, StringComparison.Ordinal);
```

Comment:
Would this fail for non-standard names of `CancellationToken`, or a fully qualified type identifier? Shouldn't it be based on the semantic model instead of the syntax model? For example, consider this:

```cs
using CT = System.Threading.CancellationToken;

class Foo {
   void SomeMethod(CT cancellationToken) { }
}
```

You would want to match on the `CT` parameter, even though its `IdentifierNameSyntax` isn't called `CancellationToken`.

Reply:
Makes sense. I'll look at its ancestors type to look for a CancellationToken type. I did something similar in a previous analyzer where I had to check if a class inherited from Stream.

Reply:
Not yet addressed in my latest commit. Still investigating.

Reply:
@AArnott thank for noticing this unhandled case, I addressed it in my latest commit - I am now using the semantic model to verify the type, and typed aliases is not a problem anymore. I also added unit tests to verify this. Can you please take a look?

Reply:
Looks great.